### PR TITLE
BUG: Fix read_json orient='table' without index (#25170)

### LINKED ISSUE #25170: Error when read_json created by to_json with orient='table and index=False
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
df = pd.DataFrame({'a': [1, 2], 'b': [3, 4]})
dfjson = df.to_json(orient='table', index=False)
result = pd.read_json(dfjson, orient='table')
```
#### Problem description
An error is raised while the original DataFrame should be recovered:
```shell
pandas/io/json/table_schema.py in parse_table_schema(json, precise_float)
    315     df = df.astype(dtypes)
    316 
--> 317     df = df.set_index(table['schema']['primaryKey'])
    318     if len(df.index.names) == 1:
    319         if df.index.name == 'index':

KeyError: 'primaryKey'
```

My pandas version is 0.24.0.

#### Expected Output
The expected output `result` should be the original DataFrame `df`.

#### Output of ``pd.show_versions()``

<details>
pandas/util/_print_versions.py in <lambda>(mod)
     78         ("blosc", lambda mod: mod.__version__),
     79         ("bottleneck", lambda mod: mod.__version__),
---> 80         ("tables", lambda mod: mod.__version__),
     81         ("numexpr", lambda mod: mod.__version__),
     82         ("feather", lambda mod: mod.__version__),

AttributeError: module 'tables' has no attribute '__version__'
</details>

### PR DESCRIPTION
**Pull request by albertvillanova**
- [x] closes #25170
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/json/test_pandas.py [hunk 1]
```diff
@@ -1262,3 +1262,12 @@ def test_index_false_error_to_json(self, orient):
...
             df.to_json(orient=orient, index=False)
+
+    @pytest.mark.parametrize('orient', ['split', 'table'])
+    def test_index_false_from_json_to_json(self, orient):
+        # GH25170
+        # Test index=False in from_json to_json
+        df = DataFrame({'a': [1, 2], 'b': [3, 4]})
+        dfjson = df.to_json(orient=orient, index=False)
+        result = read_json(dfjson, orient=orient)
+        assert_frame_equal(result, df)
```
Comment (WillAyd) [inline pandas/tests/io/json/test_pandas.py]: Usually like to do `assert_frame_equal(result, expected)` so if you could update naming conventions here would be great

File: pandas/tests/io/json/test_pandas.py [hunk 2]
```diff
@@ -1262,3 +1262,12 @@ def test_index_false_error_to_json(self, orient):
                "'orient' is 'split' or 'table'")
         with pytest.raises(ValueError, match=msg):
             df.to_json(orient=orient, index=False)
+
+    @pytest.mark.parametrize('orient', ['split', 'table'])
+    def test_index_false_from_json_to_json(self, orient):
```
Comment (WillAyd) [inline pandas/tests/io/json/test_pandas.py]: Do we have adequate roundtrip coverage on `index=True` as well? Might not hurt to expand the parametrization here if not already covered

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR! Will ultimately want a whatsnew note for v0.24.2 as well

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatsnew note, 0.24.2 is ok, in the bug fixes I/O section.

File: pandas/tests/io/json/test_pandas.py (hunk 2 above)
Comment (albertvillanova) [reply pandas/tests/io/json/test_pandas.py]: I have added parametrization test also for `index=True`.

**Comment by albertvillanova** [comment]
I have addressed all the requested changes.

File: doc/source/whatsnew/v0.24.2.rst [hunk 3]
```diff
@@ -50,6 +50,7 @@ Bug Fixes
 **I/O**
 
 - Bug in reading a HDF5 table-format ``DataFrame`` created in Python 2, in Python 3 (:issue:`24925`)
+- KeyError raised by :func:`parse_table_schema` when reading a JSON with ``orient='table'`` generated by :meth:`DataFrame.to_json` with ``index=False``
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.2.rst]: Bug in .......

you don't need to say KeyError, nor mention the parse_table_schema which is internal

add the issue number here

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, otherwise ping on green.

**Review by WillAyd — APPROVED** [review]
lgtm. Test failure I don't think is unrelated but if you can merge master and push ping on green

File: doc/source/whatsnew/v0.24.2.rst (hunk 3 above)
Comment (albertvillanova) [reply doc/source/whatsnew/v0.24.2.rst]: Done @jreback

**Comment by albertvillanova** [comment]
@WillAyd @jreback I have done all requested changes and tests pass after merging master branch.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
rebased

**Comment by jreback** [comment]
thanks @albertvillanova

**Pull request merged**